### PR TITLE
dcnm_module_utils: DCNM fabric inventory {ip:serial}

### PR DESCRIPTION
**Pull request by nkshrishail**
##### SUMMARY
Creating dcnm module util file structure and moving a common utility to module_utils

##### ISSUE TYPE
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME
dcnm

##### ADDITIONAL INFORMATION
The method in module_utils for dcnm will return fabric inventory details in the form of a dict
{
    "10.122.197.222": "9WNK9WRKCD9",
    "10.122.197.224": "99SBF117MPC",
    "10.122.197.225": "947BSAMCAU1"
}

### CODE REVIEW

File: lib/ansible/module_utils/network/dcnm/dcnm.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    for device in response:
+        ip_addr = device.get('ipAddress')
+        serial_number = device.get('serialNumber')
+        all_ip_serial.update({ip_addr: serial_number})
+
+    return all_ip_serial
+
+
+def dcnm_connection(module, method, path, json_data=None):
```
Comment (mikewiebe) [inline lib/ansible/module_utils/network/dcnm/dcnm.py]: You created this to make it easier to add unit tests right?  Do your unit tests mocks need to change as a result of moving this in?

File: lib/ansible/module_utils/network/dcnm/dcnm.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    response = dcnm_connection(module, method, path)
+
+    if response and isinstance(response, list):
+        if response[0].get('ERROR') == 'Not Found' and \
+                response[0].get('RETURN_CODE') == 404:
+                return {}
+
+    for device in response:
+        ip_addr = device.get('ipAddress')
+        serial_number = device.get('serialNumber')
```
Comment (chrisvanheuveln) [inline lib/ansible/module_utils/network/dcnm/dcnm.py]: Nit: I'm usually all about more descriptive var names but in this case I'd be happy with some shorthand, e.g. `ip` and `sn`. :-)

File: lib/ansible/module_utils/network/dcnm/dcnm.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#
+
+from ansible.module_utils.connection import Connection
+
+def get_fabric_inventory_details(module, fabric):
+    method = 'GET'
+    path = '/rest/control/fabrics/{}/inventory'.format(fabric)
+
+    all_ip_serial = dict()
+    response = dcnm_connection(module, method, path)
```
Comment (chrisvanheuveln) [inline lib/ansible/module_utils/network/dcnm/dcnm.py]: Nit: another minor style comment... this call target is actually for a `send` action, while `dcnm_connection` should refer to the conn object itself. It makes more sense to me to call the method `send`, or `dcnm_send` instead.

edit: Neither of my comments are blockers.

File: lib/ansible/module_utils/network/dcnm/dcnm.py (hunk 1 above)
Comment (nkshrishail) [reply lib/ansible/module_utils/network/dcnm/dcnm.py]: Oh yeah, I need to touch up the units before I can push this. Thanks for catching it.

File: lib/ansible/module_utils/network/dcnm/dcnm.py (hunk 2 above)
Comment (nkshrishail) [reply lib/ansible/module_utils/network/dcnm/dcnm.py]: Ok, I will try to keep you happy.

File: lib/ansible/module_utils/network/dcnm/dcnm.py (hunk 3 above)
Comment (nkshrishail) [reply lib/ansible/module_utils/network/dcnm/dcnm.py]: dcnm_send makes sense

**Review by mikewiebe — APPROVED** [review]

**Pull request merged**